GitSearch/Models/SearchItem.swift

```diff
@@ -13,56 +13,57 @@ class SearchItem: Decodable {
     let name: String?
     let score: Double
     let owner: Owners?
-    let created_at: String?
-    let updated_at: String?
+    let created: CustomDateJSON<RFC5322Date>
+    let updated: CustomDateJSON<RFC5322Date>
     let language: String?
-    let html_url: String?
+    let url: String?
     let description: String?
+    
+    enum CodingKeys: String, CodingKey {
+        case id, name, score, owner, language, description
+        case created = "created_at"
+        case updated = "updated_at"
+        case url = "html_url"
+    }
 
-    func ToArray(props: inout [String]) -> Dictionary<String, String> {
-        var result = [String: String]()
-        var childValue: String?
-        let propsFilled = props.count != 0
-        
-        let dateFormatterGet = DateFormatter()
-        dateFormatterGet.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZ"
-        
-        let dateFormatter = DateFormatter()
-        dateFormatter.dateFormat = "MMM dd,yyyy"
+    struct Owners: Decodable {
+        let login: String?
+        let id: Int
+        let url: String?
         
-        let mirror = Mirror(reflecting: self)
-        for child in mirror.children  {
-            let childKey = child.label!
-            if childKey != "owner" {
-                if let value = child.value as? Int {
-                    childValue = String(value)
-                }
-                else if let value = child.value as? Double {
-                    childValue = String(value)
-                }
-                else if let value = child.value as? String {
-                    if let date: Date = dateFormatterGet.date(from: value) {
-                        childValue = dateFormatter.string(from: date)
-                    } else {
-                        childValue = value
-                    }
-                }
-                else {childValue = ""}
-            } else {
-                childValue = (child.value as! Owners).login!
-            }
-            if !propsFilled {
-                props.append(childKey)
-            }
-            result[childKey] = childValue
+        enum CodingKeys: String, CodingKey {
+            case login, id, url = "html_url"
         }
-        return result
     }
-    
 }
 
-struct Owners: Decodable {
-    let login: String?
-    let id: Int
-    let html_url: String?
+extension SearchItem {
+    
+    static let maxPropertiesToDisplay: Int = 7 // last one is description
+    
+    subscript(index: Int) -> (name: String, value: String) {
+        switch index {
+        case 0:
+            return (NSLocalizedString("search-item.name", tableName: nil, bundle: .main, value: "name", comment: "repo's name"), self.name ?? "")
```

Comment:
`SearchItem` is [Model](https://developer.apple.com/library/archive/documentation/General/Conceptual/DevPedia-CocoaCore/MVC.html). This is very important, very fundamental patterns that a lot of things is built upon it, so I recommend read also more detailed explanations [here](https://developer.apple.com/library/archive/documentation/General/Conceptual/CocoaEncyclopedia/Model-View-Controller/Model-View-Controller.html#//apple_ref/doc/uid/TP40010810-CH14-SW1).

In this particular case, problem is that you model combines date that relates more to the presentation (view) layer. This should be separated. You can, for example, create separate helper wrapper that will encapsulated information specific to UI (like localised strings).